pkg/kubelet/api/v1alpha1/runtime/api.pb.go

```diff
@@ -0,0 +1,2867 @@
+}
+func (x *Protocol) UnmarshalJSON(data []byte) error {
+	value, err := proto.UnmarshalJSONEnum(Protocol_value, data, "Protocol")
+	if err != nil {
+		return err
+	}
+	*x = Protocol(value)
+	return nil
+}
+
+type PodSandBoxState int32
+
+const (
+	PodSandBoxState_READY    PodSandBoxState = 0
+	PodSandBoxState_NOTREADY PodSandBoxState = 1
+)
+
+var PodSandBoxState_name = map[int32]string{
+	0: "READY",
+	1: "NOTREADY",
+}
+var PodSandBoxState_value = map[string]int32{
+	"READY":    0,
+	"NOTREADY": 1,
+}
+
+func (x PodSandBoxState) Enum() *PodSandBoxState {
+	p := new(PodSandBoxState)
+	*p = x
+	return p
+}
+func (x PodSandBoxState) String() string {
+	return proto.EnumName(PodSandBoxState_name, int32(x))
+}
+func (x *PodSandBoxState) UnmarshalJSON(data []byte) error {
+	value, err := proto.UnmarshalJSONEnum(PodSandBoxState_value, data, "PodSandBoxState")
+	if err != nil {
+		return err
+	}
+	*x = PodSandBoxState(value)
+	return nil
+}
+
+type ContainerState int32
+
+const (
+	ContainerState_CREATED ContainerState = 0
+	ContainerState_RUNNING ContainerState = 1
+	ContainerState_EXITED  ContainerState = 2
+	ContainerState_UNKNOWN ContainerState = 3
+)
+
+var ContainerState_name = map[int32]string{
+	0: "CREATED",
+	1: "RUNNING",
+	2: "EXITED",
+	3: "UNKNOWN",
+}
+var ContainerState_value = map[string]int32{
+	"CREATED": 0,
+	"RUNNING": 1,
+	"EXITED":  2,
+	"UNKNOWN": 3,
+}
+
+func (x ContainerState) Enum() *ContainerState {
+	p := new(ContainerState)
+	*p = x
+	return p
+}
+func (x ContainerState) String() string {
+	return proto.EnumName(ContainerState_name, int32(x))
+}
+func (x *ContainerState) UnmarshalJSON(data []byte) error {
+	value, err := proto.UnmarshalJSONEnum(ContainerState_value, data, "ContainerState")
+	if err != nil {
+		return err
+	}
+	*x = ContainerState(value)
+	return nil
+}
+
+type VersionRequest struct {
+	// The version of kubelet runtime API.
+	Version          *string `protobuf:"bytes,1,opt,name=version" json:"version,omitempty"`
+	XXX_unrecognized []byte  `json:"-"`
+}
+
+func (m *VersionRequest) Reset()         { *m = VersionRequest{} }
+func (m *VersionRequest) String() string { return proto.CompactTextString(m) }
+func (*VersionRequest) ProtoMessage()    {}
+
+func (m *VersionRequest) GetVersion() string {
+	if m != nil && m.Version != nil {
+		return *m.Version
+	}
+	return ""
+}
+
+type VersionResponse struct {
+	// The version of the kubelet runtime API.
+	Version *string `protobuf:"bytes,1,opt,name=version" json:"version,omitempty"`
+	// The name of the container runtime.
+	RuntimeName *string `protobuf:"bytes,2,opt,name=runtime_name" json:"runtime_name,omitempty"`
+	// The version of the container runtime.
+	RuntimeVersion *string `protobuf:"bytes,3,opt,name=runtime_version" json:"runtime_version,omitempty"`
+	// The API version of the container runtime.
+	RuntimeApiVersion *string `protobuf:"bytes,4,opt,name=runtime_api_version" json:"runtime_api_version,omitempty"`
```

Comment:
Right thanks @feiskyer, but, why do you need information so deep in the runtime interface? I cannot think of anything